Add functionality to import data from Instagram

luxuc opened the pull request:
Get a list of Instagram data near Troy and return the list in getPost.
Sort superfeed posts and Insta posts 
Display posts in feed.

luxuc committed:
Add functionality to import data from Instagram

rrdelaney commented on src/components/Insta.jsx:
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+export default function Insta ({ avator, name, body, image }) {
```
The correct spelling is `avatar`. Please correct that throughout the PR

rrdelaney commented on src/functions/getPosts/index.js:
```diff
@@ -47,6 +48,31 @@ function getTweets ({latitude, longitude}) {
+function getIns (location) {
+  let ig = i.instagram()
+
+  ig.use({ access_token: '3178355401.1677ed0.117cca9f55c847018b161e79287dcac3' })
```
The access token should be an environment variable

rrdelaney commented on src/functions/getPosts/index.js:
```diff
@@ -47,6 +48,31 @@ function getTweets ({latitude, longitude}) {
+  return new Promise(function (resolve, reject) {
+    let options = {}
+    ig.location_media_recent(location, [options], function (err, result, pagination, remaining, limit) {
+      if (err) reject(console.log(err))
```
Errors should be sent to STDERR using `console.error`

rrdelaney commented on src/functions/getPosts/index.js:
```diff
@@ -57,7 +83,9 @@ export const handler = async function (e) {
   })
 
-  let results = posts.concat(tweets).sort((a, b) => (new Date(b.created) - new Date(a.created)))
+  let ins = await getIns('219471895') // it is location_id of troy
```
Lookup should happen with Lat/Long. This might require another API call

rrdelaney commented:
Overall looks good. Some spelling errors, and the things I commented on. After that it should be good to merge. Please don't close this pull when you push new changes. Just push to the same branch and the changes will reflect here.

luxuc committed:
Change spelling error 'avator' to 'avatar'
Make Instagram access_token a process  environment variable
Change the way to output error
Make a getLocationId function which calls /locations/search endpoint. It will return a list of locations ordered by the proximity with lat&lng.
Use the first (closest) location id in the list.

rrdelaney commented:
LGTM 👍
